### ISSUE TITLE
YAML files are ignored and treated as local refs

### ISSUE THREAD

**Issue by kobik** [cvent/json-schema-deref-sync#28]
1) What version of the module is the issue happening on? Does the issue happen on latest version?
0.10.1

2) What platform and Node.js version? (For example Node.js 0.12 on Mac OS X)
NodeJS v10.16.0 on MacOS Mojave

3) Sample source code or steps to reproduce
Use YAML file refs
```json
$ref: "./validator/countries.yaml#/routes/base"
```
[countries.yaml.zip](https://github.com/cvent/json-schema-deref-sync/files/3596023/countries.yaml.zip)

(Write description of your issue here, stack traces from errors and code that reproduces the issue are helpful)
YAML files are ignored and treated as local refs, as a result I'm getting this stack trace
```stack
TypeError: Cannot read property 'countries.yaml#' of undefined
  at /home/api/node_modules/json-schema-deref-sync/lib/utils.js:84:17
  at Array.reduce (<anonymous>)
  at Object.getRefPathValue (/home/api/node_modules/json-schema-deref-sync/lib/utils.js:83:15)
  at getRefSchema (/home/api/node_modules/json-schema-deref-sync/lib/index.js:99:18)
  at Object.<anonymous> (/home/api/node_modules/json-schema-deref-sync/lib/index.js:236:24)
  at walker (/home/api/node_modules/traverse/index.js:190:22)
  at /home/api/node_modules/traverse/index.js:208:29
  at Array.forEach (<anonymous>)
  at forEach (/home/api/node_modules/traverse/index.js:298:31)
  at walker (/home/api/node_modules/traverse/index.js:203:13)
  at /home/api/node_modules/traverse/index.js:208:29
  at Array.forEach (<anonymous>)
  at forEach (/home/api/node_modules/traverse/index.js:298:31)
  at walker (/home/api/node_modules/traverse/index.js:203:13)
  at walk (/home/api/node_modules/traverse/index.js:226:7)
  at Traverse.forEach (/home/api/node_modules/traverse/index.js:50:18)
  at derefSchema (/home/api/node_modules/json-schema-deref-sync/lib/index.js:223:27)
  at deref (/home/api/node_modules/json-schema-deref-sync/lib/index.js:310:13)
  at Object.buildSchemaSync (/home/api/node_modules/api-schema-builder/src/index.js:35:26)
  at Object.init (/home/api/node_modules/express-ajv-swagger-validation/src/middleware.js:25:32)
  at Object.<anonymous> (/home/api/dist/server.js:21:28)
  at Generator.next (<anonymous>)
```

I already have a PR ready to fix this issue https://github.com/cvent/json-schema-deref-sync/pull/27

**Comment by jcurtis** [comment]
Hey @kobik 

This package is specifically for JSON schemas, not YAML. Could you use `options.loaders` to implement a custom loader to fix your use case instead of extending this codebase? 

Here's the PR where that got implemented: https://github.com/cvent/json-schema-deref-sync/pull/22
Here's the default JSON loader: https://github.com/cvent/json-schema-deref-sync/blob/master/lib/loaders/file.js

**Comment by kobik** [comment]
@jcurtis, so I've made few changes to my PR.

1. Removed my YAML loader addition
2. Fixed getRefType function as it wasn't mapping types correctly
See this [Structuring a complex schema](https://json-schema.org/understanding-json-schema/structuring.html#reuse)
And specifically this
> If you’re using a definition from the same document, the $ref value begins with the pound symbol (#). Following that, the slash-separated items traverse the keys in the objects in the document. 
3. Added test for validating loading of YAML files using custom loader
4. Renamed the title to a one better describing the issue it fixes

**Comment by kobik** [comment]
@jcurtis, can you guys please publish the latest version to npm?

**Comment by jcurtis** [comment]
@kobik Unfortunately I'm still having issues securing the permission to publish to npm. You may have better luck forking the repo and publishing your own version. I apologize for the inconvenience.

**Comment by bojand** [comment]
I just published `0.13.0`; hope that was OK. Although couldn't tag it here as it seems I do not have permissions to push tags to this repo any more.

@jcurtis I added you to the NPM package a few days ago. See  [profile](https://www.npmjs.com/~jcurtis) You should be able to publish? What's the issue?
